### PR TITLE
Move policies to internal

### LINKED ISSUE #1882: Policies (not their options) are public. Shoud they be _internal?

### PR DESCRIPTION
**Pull request by antkmsft**
Closes #1882.

### CODE REVIEW

**Review by Jinming-Hu — APPROVED** [review]
Changes to storage LGTM

File: sdk/core/azure-core/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 ## 1.0.0-beta.8 (Unreleased)
 
+### Breaking Changes
+
+- Removed from `Azure::Core::Http::Policies` namespace: `HttpPolicy`, `NextHttpPolicy`, `HttpPolicyOrder`, `TransportPolicy`, `RetryPolicy`, `RequestIdPolicy`, `TelemetryPolicy`, `BearerTokenAuthenticationPolicy`, `LogPolicy`.
```
Comment (ahsonkhan) [inline sdk/core/azure-core/CHANGELOG.md:7]: `HttpPolicy` which is the base class for all policies needs to remain public.

**Review by ahsonkhan — CHANGES_REQUESTED** [review]
`HttpPolicy` which is the base class for all policies needs to remain public.

File: sdk/core/azure-core/inc/azure/core/internal/client_options.hpp [hunk 2]
```diff
@@ -29,13 +29,15 @@ namespace Azure { namespace Core { namespace _internal {
      * @brief Define policies to be called one time for every Http request from an sdk client.
      *
      */
-    std::vector<std::unique_ptr<Azure::Core::Http::Policies::HttpPolicy>> PerOperationPolicies;
+    std::vector<std::unique_ptr<Azure::Core::Http::Policies::_internal::HttpPolicy>>
+        PerOperationPolicies;
```
Comment (ahsonkhan) [inline sdk/core/azure-core/inc/azure/core/internal/client_options.hpp:33]: Revert the change to this file

File: sdk/core/azure-core/inc/azure/core/internal/http/pipeline.hpp [hunk 3]
```diff
@@ -37,8 +37,8 @@ namespace Azure { namespace Core { namespace Http { namespace _internal {
     /**
      * @brief Construct HTTP pipeline with the sequence of HTTP policies provided.
      *
-     * @param policies A sequence of #Azure::Core::Http::Policies::HttpPolicy representing a stack,
-     * first element corresponding to the top of the stack.
+     * @param policies A sequence of #Azure::Core::Http::Policies::HttpPolicy
+     * representing a stack, first element corresponding to the top of the stack.
```
[the remark this answers is not in the document]
Comment (antkmsft) [reply sdk/core/azure-core/inc/azure/core/internal/http/pipeline.hpp:41]: Ctrl-K-D produced that :)

[the remark this answers is not in the document]
Comment (vhvb1989) [reply sdk/core/azure-core/inc/azure/core/internal/http/pipeline.hpp:41]: not sure. It might be a diff clang format version but still compatible with version 10.  Not sure.
If we start looking  changes to this file from one developer to another, then we need to investigate further
